dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py

```diff
@@ -0,0 +1,753 @@
+        "SIC_2_digit"
+    ].apply(lambda x: sic_names.get(x) if x else None)
+
+    return all_green_measures_df
+
+
+def get_soc_info():
+    # we need soc descriptions as well - they have descriptions for 6 digit sic codes
+    soc_descriptions = pd.read_excel(
+        "s3://prinz-green-jobs/inputs/data/occupation_data/ons/extendedsoc2020structureanddescriptionsexcel121023.xlsx",
+        sheet_name="Extended SOC descriptions MG1-9",
+    )
+    soc_descriptions.columns = [
+        i.lower().strip().replace(" ", "_").replace("-", "")
+        for i in soc_descriptions.iloc[0].values
+    ]
+    soc_descriptions.drop(0, inplace=True)
+    soc_descriptions = (
+        soc_descriptions[["subunit_group", "group_title", "descriptions"]]
+        # drop na if its na in any of the columns
+        .dropna(subset=["subunit_group", "group_title", "descriptions"])
+    )
+    # clean description
+    soc_descriptions["clean_description"] = soc_descriptions.descriptions.apply(
+        lambda x: x.replace("\n", " ")
+        .replace("\r", " ")
+        .replace("\t", " ")
+        .replace("  ", " ")
+        .strip()
+        .replace("\xa0", "")
+    )
+    soc_descriptions_dict = soc_descriptions.set_index(
+        "subunit_group"
+    ).clean_description.to_dict()
+
+    return soc_descriptions_dict
+
+
+def add_green_topics(all_green_measures_df):
+    om = OccupationMeasures()
+    om.load()
+
+    all_green_measures_df["green_topics_lists"] = all_green_measures_df[
+        "SOC_2010"
+    ].apply(
+        lambda x: om.soc_green_measures_dict.get(x)["ONET_green_topics"]
+        if x in om.soc_green_measures_dict
+        else None
+    )
+
+    return all_green_measures_df
+
+
+def add_additional_metadata(all_green_measures_df, job_id_col="job_id"):
+    all_green_measures_df = add_salaries(all_green_measures_df, job_id_col=job_id_col)
+
+    all_green_measures_df = add_locations(all_green_measures_df, job_id_col=job_id_col)
+
+    all_green_measures_df = add_sic_info(all_green_measures_df)
+
+    soc_descriptions_dict = get_soc_info()
+
+    all_green_measures_df = add_green_topics(all_green_measures_df)
+
+    return all_green_measures_df, soc_descriptions_dict
+
+
+def filter_large_occs(
+    all_green_measures_df, min_num_job_ads=50, occ_col="SOC_2020_name"
+):
+    # get occupations for which we have over 50 job adverts for
+    representative_occs = (
+        all_green_measures_df.groupby(occ_col)
+        .job_id.count()
+        .sort_values(ascending=False)
+        .where(lambda x: x >= min_num_job_ads)
+        .dropna()
+        .keys()
+        .tolist()
+    )
+
+    filtered_df = all_green_measures_df[
+        all_green_measures_df[occ_col].isin(representative_occs)
+    ].reset_index(drop=True)
+
+    print(f"Filtered from {len(all_green_measures_df)} rows to {len(filtered_df)}")
+
+    return filtered_df
+
+
+def create_green_skills_df(all_green_measures_df, occ_col="SOC_2020_name"):
```

Comment:
docstrings + type hints

Reply:
i've realised there are so many functions actually - i don't know how important docstrings + type hints are at this point given the analysis is very live and updating - maybe we just make sure to save a few days to clean up the analysis code once we're happy with the graphs

Reply:
I've added the type hints throughout. but I think docstrings could just be left for the most important functions rather than all of them. I agree, it feels a bit in flux to spend too much time on.